### PR TITLE
Create M5Station pins_arduino.h and M5Station Board

### PR DESCRIPTION
**Pull request by Tinyu-Zhao**
M5Station related [documents](https://docs.m5stack.com/en/core/station_485)

### CODE REVIEW

**Comment by Tinyu-Zhao** [comment]
@SuGlider Please help with this merge request.🤖

File: boards.txt [hunk 1]
```diff
@@ -9895,6 +9895,91 @@ m5stack-fire.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-station.menu.DebugLevel.error.build.code_debug=1
+m5stack-station.menu.DebugLevel.warn=Warn
+m5stack-station.menu.DebugLevel.warn.build.code_debug=2
+m5stack-station.menu.DebugLevel.info=Info
+m5stack-station.menu.DebugLevel.info.build.code_debug=3
+m5stack-station.menu.DebugLevel.debug=Debug
+m5stack-station.menu.DebugLevel.debug.build.code_debug=4
+m5stack-station.menu.DebugLevel.verbose=Verbose
+m5stack-station.menu.DebugLevel.verbose.build.code_debug=5
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:9980]: Can you please add the Erase_Flash menu option, its needed for all boards from PR #7043 and PR #7129 is adding that to all existing boards.

Comment (Tinyu-Zhao) [reply boards.txt:9980]: Thank you for your feedback, I will add in time.

Comment (Tinyu-Zhao) [reply boards.txt:9980]: Already taken care of

File: boards.txt [hunk 2]
```diff
@@ -9895,6 +9895,96 @@ m5stack-fire.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-station.bootloader.tool=esptool_py
+m5stack-station.bootloader.tool.default=esptool_py
+
+m5stack-station.upload.tool=esptool_py
+m5stack-station.upload.tool.default=esptool_py
+m5stack-station.upload.tool.network=esp_ota
+
+m5stack-core2.upload.maximum_size=6553600
+m5stack-core2.upload.maximum_data_size=4521984
```
Comment (P-R-O-C-H-Y) [inline boards.txt:9908]: Please fix the name of the board to `m5stack-station`

File: boards.txt (hunk 1 above)
Comment (P-R-O-C-H-Y) [reply boards.txt:9980]: Thanks

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]

File: boards.txt (hunk 2 above)
Comment (Tinyu-Zhao) [reply boards.txt]: Sorry, this has been corrected

**Review by P-R-O-C-H-Y — APPROVED** [review]
Thanks for updates. LGTM.

**Comment by Tinyu-Zhao** [comment]
Thank you for your help.

**Comment by Tinyu-Zhao** [comment]
@P-R-O-C-H-Y 

Hello, we have a strange problem, I do not know how to solve it, can you help me to see what the problem is?🙏
https://github.com/espressif/arduino-esp32/issues/7317#issuecomment-1264335731

**Pull request merged**